src/fabric/src/fabric_view_all_docs.erl

```diff
@@ -286,6 +286,20 @@ all_docs_concurrency() ->
             10
     end.
 
+validate_if_partition(Row, Acc) ->
+    QueryStr = Acc#vacc.req#httpd.qs,
+    case QueryStr of
+        undefined -> Row;
+        [{"partition", Id}] -> is_doc_in_partition(Row, Id)
```

Comment:
this only matches if the `qs` is either `undefined` or exactly the list `[{"partition", Id}]` — is this always the case? Can there be other options in the qs? If yes, we’d probably need to get the partition tuple out with a `proplists:get()` or `couch_util:get_value()`

Reply:
You are correct, I am newish to the pattern matching in erlang and here it really shows.  I can see how this will fail if there are any other query parameters.